tools/cp-caps/rcp_caps_test.py

```diff
@@ -0,0 +1,414 @@
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import argparse
+import logging
+import os
+import sys
+import textwrap
+
+from typing import List
+
+import otci
+from otci import OTCI
+
+logging.basicConfig(level=logging.WARNING)
+
+
+class RcpCaps(object):
+    """
+    This class represents an OpenThread RCP capability test instance.
+    """
+
+    def __init__(self):
+        self.__dut = self.__connect_dut()
+        self.__ref = self.__connect_reference_device()
+
+    def test_diag_commands(self):
+        """Test all diag commands."""
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        ret = self.__dut.is_command_supported('diag start')
+        if ret is False:
+            print('All diag commands are not supported')
+            return
+
+        self.__dut.diag_start()
+        self.__ref.diag_start()
+
+        self.__test_diag_channel()
+        self.__test_diag_power()
+        self.__test_diag_radio()
+        self.__test_diag_repeat()
+        self.__test_diag_send()
+        self.__test_diag_frame()
+        self.__test_diag_echo()
+        self.__test_diag_utils()
+        self.__test_diag_rawpowersetting()
+        self.__test_diag_powersettings()
+        self.__test_diag_gpio_mode()
+        self.__test_diag_gpio_value()
+
+        self.__ref.diag_stop()
+        self.__dut.diag_stop()
+
+    #
+    # Private methods
+    #
+    def __test_diag_channel(self):
+        channel = 20
+        commands = ['diag channel', f'diag channel {channel}']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_channel(channel)
+            channel = self.__dut.diag_get_channel()
+            ret = True if channel == channel else False
+        else:
+            ret = False
+
+        self.__output_results(commands, ret)
+
+    def __test_diag_power(self):
+        power = self.__get_dut_diag_power()
+        commands = ['diag power', f'diag power {power}']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_power(power)
+            value = self.__dut.diag_get_power()
+            ret = True if value == power else False
+        else:
+            ret = False
+
+        self.__output_results(commands, ret)
+
+    def __test_diag_radio(self):
+        commands = ['diag radio receive', 'diag radio sleep', 'diag radio state']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_radio_receive()
+            receive_state = self.__dut.diag_get_radio_state()
+            self.__dut.wait(0.1)
+            self.__dut.diag_radio_sleep()
+            sleep_state = self.__dut.diag_get_radio_state()
+
+            ret = True if sleep_state == 'sleep' and receive_state == 'receive' else False
```

Comment:
Will the following work?
```suggestion
            ret = sleep_state == 'sleep' and receive_state == 'receive'
```